CommonTools/PileupAlgos/src/PuppiAlgo.cc

```diff
@@ -8,143 +8,221 @@
-  if(!fAdjust[iAlgo]) return;
-  //Adjust the p-value to correspond to the median
-  std::sort(fPupsPV.begin(),fPupsPV.end());
-  int lNPV = 0; for(unsigned int i0 = 0; i0 < fPupsPV.size(); i0++) if(fPupsPV[i0] <= lMed ) lNPV++; 
-  double lAdjust = 1.5*double(lNPV)/double(fPupsPV.size()+fNCount[iAlgo]);
-  if(lAdjust > 0) fMedian[iAlgo] -= sqrt(ROOT::Math::chisquared_quantile(lAdjust,1.)*fRMS[iAlgo]);
+    // sort alphas
+    int lNBefore = 0;
+    for(unsigned int i0 = 0; i0 < iAlgo; i0++) lNBefore += fNCount[i0];
+    std::sort(fPups.begin()+lNBefore,fPups.begin()+lNBefore+fNCount[iAlgo]);
+    
+    // some correction that phil added...
+    double lCorr = 1.;
+    //if(!fCharged[iAlgo] && fAdjust[iAlgo]) lCorr *= 1. - iPVFrac;
+    if(fAdjust[iAlgo]) lCorr *= 1. - iPVFrac;
+
+    // in case you have alphas == 0
+    int lNum0 = 0;
+    for(int i0 = lNBefore; i0 < lNBefore+fNCount[iAlgo]; i0++) {
+        if(fPups[i0] == 0) lNum0 = i0-lNBefore;
+    }
+
+    // comput median, removed lCorr for now
+    int lNHalfway = lNBefore + lNum0 + int( double( fNCount[iAlgo]-lNum0 )*0.50);
+    fMedian[iAlgo] = fPups[lNHalfway];
+    double lMed = fMedian[iAlgo];  //Just to make the readability easier
+    
+    int lNRMS = 0;
+    for(int i0 = lNBefore; i0 < lNBefore+fNCount[iAlgo]; i0++) {
+        fMean[iAlgo] += fPups[i0];
+        if(fPups[i0] == 0) continue;
+        if(!fCharged[iAlgo] && fAdjust[iAlgo] && fPups[i0] > lMed) continue;
+        //if(fAdjust[iAlgo] && fPups[i0] > lMed) continue;
+        lNRMS++;
+        fRMS [iAlgo] += (fPups[i0]-lMed)*(fPups[i0]-lMed);
+    }
+    fMean[iAlgo]/=fNCount[iAlgo];
+    if(lNRMS > 0) fRMS [iAlgo]/=lNRMS;
+    if(fRMS[iAlgo] == 0) fRMS[iAlgo] = 1e-5;
+    // here is the raw RMS
+    fRMS [iAlgo] = sqrt(fRMS[iAlgo]);
+
+    // some ways to do corrections to fRMS and fMedian
+    fRMS [iAlgo] *= fRMSScaleFactor[iAlgo];
+
+    fRMS[iAlgo]    *= fRMSEtaSF;
+    fMedian[iAlgo] *= fMedEtaSF;
+    //if(!fCharged[iAlgo]) std::cout << " Process : " << iAlgo  << " Median : " << fMedian[iAlgo] << " +/- " << fRMS[iAlgo]  << " -- Begin : " << lNBefore << " -- Total :  " << fNCount[iAlgo] << " -- 50% " << lNHalfway  << " Fraction less than @ Median : " << std::endl;
+    
+    // if(!fAdjust[iAlgo]) return;
+    // //Adjust the p-value to correspond to the median
+    // std::sort(fPupsPV.begin(),fPupsPV.end());
+    // int lNPV = 0; for(unsigned int i0 = 0; i0 < fPupsPV.size(); i0++) if(fPupsPV[i0] <= lMed ) lNPV++;
+    // double lAdjust = 1.5*double(lNPV)/double(fPupsPV.size()+fNCount[iAlgo]);
+    // if(lAdjust > 0) fMedian[iAlgo] -= sqrt(ROOT::Math::chisquared_quantile(lAdjust,1.)*fRMS[iAlgo]);
 }
+////////////////////////////////////////////////////////////////////////////////
```

Comment:
unless all of this commented out code is going to be imminently necessary, please remove.
One can always go back to an older release or an older commit to make comparisons